cruize_operations/views.py

```diff
@@ -1,36 +1,83 @@
         serializer = self.serializer_class(data=request.data)
         serializer.is_valid(raise_exception=True)
         amount = serializer.data
-        cruize_contract_ref = Cruize()
         try:
-            result = "passs"
-            result = cruize_contract_ref.repay_to_aave(amount)
+            result["message"] = self.cruize_contract_ref.repay_to_aave(amount)
             return Response(result, status.HTTP_200_OK)
         except Exception as e:
-            return Response(e, status.HTTP_500_INTERNAL_SERVER_ERROR)
+            result["error"] = e
+            return Response(result, status.HTTP_500_INTERNAL_SERVER_ERROR)
 
     def deposit(self, request):
+        self.initialize()
+        result = {"message": None, "error": None}
         self.serializer_class = CruizeDepositRequestSerializer
         serializer = self.serializer_class(data=request.data)
         serializer.is_valid(raise_exception=True)
         deposit_data = serializer.data
         try:
-            cruize_contract_ref = Cruize()
-            result = cruize_contract_ref.deposit_to_cruize(deposit_data)
+            result["message"] = self.cruize_contract_ref.deposit_to_cruize(deposit_data)
+            return Response(result, status.HTTP_200_OK)
+        except Exception as e:
+            result["error"] = e
+            return Response(result, status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def save_transactions(self, request):
+        result = {"message": None, "error": None}
+        self.initialize()
+        self.serializer_class = FirebaeRequestSerializer
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        deposit_data = serializer.data
+        try:
+            self.firebase_data_manager_obj = FirebaseDataManager()
+            self.firebase_data_manager_obj.store_data(
+                deposit_data, cruize_constants.CRUIZE_USER
+            )
+            result["message"] = "success"
+            return Response(result, status.HTTP_200_OK)
+        except Exception as e:
+            result["error"] = e
+            return Response(result, status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+    def fetch_user_transactions(self, request):
+        result = {"message": None, "error": None}
+        self.initialize()
+        self.serializer_class = FirebaseFecthRequestSerializer
+        serializer = self.serializer_class(data=request.data)
+        serializer.is_valid(raise_exception=True)
+        data = serializer.data
+        try:
+            self.firebase_data_manager_obj = FirebaseDataManager()
+            result["message"] = self.firebase_data_manager_obj.fetch_transaction_data(
+                cruize_constants.CRUIZE_USER, data
+            )
             return Response(result, status.HTTP_200_OK)
         except Exception as e:
-            return Response(e, status.HTTP_500_INTERNAL_SERVER_ERROR)
+            result["error"] = e
```

Comment:
which transactions are we saving and fetching here ?

Reply:
the user's deposit and withdrawal transactions